src/inputcomponents/kickdownswitch.cpp

```diff
@@ -0,0 +1,13 @@
+#include "kickdownswitch.hpp"
+#include "board_config.h"
+
+
+bool KickdownSwitch::last_state = false; // Initialize the last state to false
+
+bool KickdownSwitch::is_kickdown_newly_pressed(EgsBaseCan *egs_can_hal, uint32_t expire_time_ms)
+{
+    bool current_state = pcb_gpio_matrix->is_kickdown_pressed() || egs_can_hal->get_kickdown(expire_time_ms);
```

Comment:
I think, we should select which input to use based on TCU Config....I am not sure if cars with TRRS have the kickdown CAN bit set for something else or not

Reply:
Must correct yesterday's statement: On EGS51 and EGS52 the TCU sends the signal. Hence, no implementation of these functions is required for EGS51-layer or EGS52-layer.
I'll delete the "TODO"-hints respectively in the next PR.